Add Waveshare RP2040-LoRa board

lindenaar opened the pull request:
I added the definitions for the [Wareshare's RP2040-LoRa](https://www.waveshare.com/wiki/RP2040-LoRa). This board is a variant of the Waveshare RP2040 One module with an integrated SX1262
 
**notes**:
 - this board uses custom SPI pins --> must call RadioBeginSPI() before use
 - this board has an antenna switch --> to enable/use call RadioEnableRfSwitch(radio)
 - this board uses an XTAL (crystal oscillator that is not temperature  compensated), which is not the default of RadioLib. The SPI initialisation will correct this setting as well (could not find a better place for that), see also the [troubleshooting page of the RadioLib Wiki](https://github.com/jgromes/RadioLib/wiki/Troubleshooting-Guide#sx126xsx128xlr11x0-incorrect-oscillator)
 
Tested with the EU (868 Mhz) model but should work for others as well

Commit:
added the Waveshare RP2040-LoRa board definition

Commit:
fixed minor typo

jgromes commented:
Looking good, just two minor things:

1. Could you add `RADIO_BOARD_WAVESHARE_RP2040_LORA` to `keywords.txt` as `LITERAL1`? Please keep in mind that the file uses true tabs, not spaces.
2. There seems to be a typo in comment in the header file (686 Mhz), it should probably be 868 MHz.

Commit:
- fixed switch state (must be HIGH for sending)
- updated comments to clarify/correct things

Commit:
added RADIO_BOARD_WAVESHARE_RP2040_LORA as literal

lindenaar commented:
@jgromes thanks for your feedback, I found and fixed and issue with the RF switch state and also fixed these:
> 1. Could you add `RADIO_BOARD_WAVESHARE_RP2040_LORA` to `keywords.txt` as `LITERAL1`? Please keep in mind that the file uses true tabs, not spaces.
> 2. There seems to be a typo in comment in the header file (686 Mhz), it should probably be 868 MHz.

please let me know in case you need any other updates.

Perhaps good to add the steps needed to contribute in the README.md or add a CONTRIBUTING.md for that? I am more than happy to provide a PR for that as well if you are open for that.

jgromes commented:
Merged, thank you for the contribution!

> Perhaps good to add the steps needed to contribute in the README.md or add a CONTRIBUTING.md for that?

Indeed, that would be very useful. I'll take care of it :)
